new feature: remove a specific song from somewhere in the queue

CArchetype opened the pull request:
I tried to match the feel of the other commands. I also made some pull requests that I had to close because I've never used GitHub before half an hour ago. Oops!

This removes a song from somewhere in the queue, tells you what song was removed, and doesn't take bad input.

Commit:
added cmd_remove

Commit:
add remove_entry for cmd_remove

Commit:
fixed consistency in !help comment for cmd_remove

Commit:
added logic to catch rare index error

sometimes you can remove a song from the very end of the queue while the player loads a new song which causes the index to no longer exist. let me know if there is a better way to handle this.

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1702,6 +1702,47 @@ def check(m):
+        # input, we know there's a song there
+        # if the user inputs 0, tell them to use skip instead
+        if not index:
+            raise exceptions.CommandError("No input found. Try again with a position from {command_prefix}queue.", expire_in=20)
```
You need to format `{command_prefix}` yourself here, the exception will not format it for you.

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1702,6 +1702,47 @@ def check(m):
                 delete_after=20
             )
 
+    async def cmd_remove(self, message, player, index=None):
```
`index` should be given as a positional argument, not a kwarg. It is a required argument for the command to work, right?

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1702,6 +1702,47 @@ def check(m):
+                raise exceptions.CommandError("You can't remove the current song (skip it instead), or a song in a position that doesn't exist.", expire_in=20)
+
+        else:
+            raise exceptions.CommandError("There is either no queue, or the queue is not that long.", expire_in=20)
```
Maybe it's worth checking if there are entries in the queue separately above this entire `if/else`.

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1702,6 +1702,47 @@ def check(m):
+            raise exceptions.CommandError('{} is not a valid number.'.format(index), expire_in=20)
+
+        if index <= len(player.playlist.entries):
+            if index > 0:
```
Why not group these two `if` statements onto the same line?

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1702,6 +1702,47 @@ def check(m):
+                except IndexError:
+                    raise exceptions.CommandError("Something went wrong. Please try again.", expire_in=20)
+                
+                return Response(':white_check_mark: **' + song_title + '** removed.', delete_after=20)
```
Copy the way we embed emojis like that in strings: https://github.com/SexualRhinoceros/MusicBot/blob/review/musicbot/bot.py#L1035

Commit:
many fixes to cmd_remove

took all suggestions from review of the PR. the command is smaller and more efficient, and also follows the same embed style for emojis.
